### PR TITLE
Update to latest Stardust changes

### PR DESCRIPTION
**Pull request by alexsporn**

### CODE REVIEW

File: plugins/restapi/v2/messages.go [hunk 1]
```diff
@@ -190,13 +190,21 @@ func sendMessage(c echo.Context) (*messageCreatedResponse, error) {
 		}
 	}
 
-	if msg.NetworkID == 0 && msg.Nonce != 0 {
+	if msg.ProtocolVersion == 0 && msg.Nonce != 0 {
```
Comment (muXxer) [inline plugins/restapi/v2/messages.go]: I guess we should check at the beginning if the message has the correct protocol version.
This must be filled with the correct value.

File: plugins/restapi/v2/messages.go [hunk 2]
```diff
@@ -190,13 +190,21 @@ func sendMessage(c echo.Context) (*messageCreatedResponse, error) {
...
 		// Message was PoWed without the correct networkId being set, so reject it
-		return nil, errors.WithMessage(restapi.ErrInvalidParameter, "invalid message, error: PoW done but networkId missing")
+		return nil, errors.WithMessage(restapi.ErrInvalidParameter, "invalid message, error: PoW done but protocolVersion missing")
 	}
 
-	if msg.NetworkID == 0 {
-		msg.NetworkID = deps.NetworkID
+	if msg.ProtocolVersion == 0 {
```
Comment (muXxer) [inline plugins/restapi/v2/messages.go]: Should be filled by the client with correct value already, no?

File: plugins/restapi/v2/messages.go [hunk 3]
```diff
@@ -190,13 +190,21 @@ func sendMessage(c echo.Context) (*messageCreatedResponse, error) {
...
+		msg.ProtocolVersion = iotago.ProtocolVersion
+	}
+
+	switch payload := msg.Payload.(type) {
+	case *iotago.Transaction:
+		if payload.Essence.NetworkID != deps.NetworkID {
+			return nil, errors.WithMessagef(restapi.ErrInvalidParameter, "invalid message, error: wrong networkID: %d", payload.Essence.NetworkID)
```
Comment (muXxer) [inline plugins/restapi/v2/messages.go]:
```suggestion
			return nil, errors.WithMessagef(restapi.ErrInvalidParameter, "invalid payload, error: wrong networkID: %d", payload.Essence.NetworkID)
```

File: pkg/model/coordinator/milestones.go [hunk 4]
```diff
@@ -14,9 +14,8 @@ import (
 // createCheckpoint creates a checkpoint message.
 func (coo *Coordinator) createCheckpoint(parents hornet.MessageIDs) (*storage.Message, error) {
 	iotaMsg := &iotago.Message{
-		NetworkID: coo.networkID,
-		Parents:   parents.ToSliceOfArrays(),
-		Payload:   nil,
+		Parents: parents.ToSliceOfArrays(),
```
Comment (muXxer) [inline pkg/model/coordinator/milestones.go]:
```suggestion
		ProtocolVersion: iotago.ProtocolVersion,
		Parents: parents.ToSliceOfArrays(),
```

Comment (muXxer) [reply pkg/model/coordinator/milestones.go]: Damn, don't forget the formatting after applying that suggestion :D

File: pkg/indexer/indexer.go [hunk 5]
```diff
@@ -304,15 +304,15 @@ func processOutput(output *utxo.Output, tx *gorm.DB) error {
...
+			FoundryID:        foundryID[:],
+			OutputID:         make(outputIDBytes, iotago.OutputIDLength),
+			NativeTokenCount: len(iotaOutput.NativeTokens),
+			CreatedAt:        unixTime(output.MilestoneTimestamp()),
 		}
 		copy(foundry.OutputID, output.OutputID()[:])
 
 		if addressUnlock := conditions.Address(); addressUnlock != nil {
-			foundry.Address, err = addressBytesForAddress(addressUnlock.Address)
+			foundry.AliasAddress, err = addressBytesForAddress(addressUnlock.Address)
```
Comment (muXxer) [inline pkg/indexer/indexer.go]: is that correct for the foundry? shouldn't you use `Indent` instead?

File: pkg/model/utxo/database_prefixes.go [hunk 6]
```diff
@@ -4,7 +4,7 @@ const (
 	UTXOStoreKeyPrefixLedgerMilestoneIndex byte = 0
 
 	// Output and Spent storage
-	UTXOStoreKeyPrefixOutput byte = 1 //TODO: iterate over all values and map to extended outputs
+	UTXOStoreKeyPrefixOutput byte = 1 //TODO: iterate over all values and map to basic outputs
```
Comment (alexsporn) [inline pkg/model/utxo/database_prefixes.go:7]: Don't get your question. This is for a possible migration from chpt2 -> stardust

**Review by muXxer — APPROVED** [review]

**Pull request merged**